src/samplers/string.js

```diff
@@ -105,7 +105,7 @@ const stringFormats = {
 };
 
 export function sampleString(schema) {
-    let format = schema.pattern !== undefined ? 'pattern' : schema.format || 'default';
+    let format = !isBlank(schema.pattern) ? 'pattern' : schema.format || 'default';
```

Comment:
You don't need to use `isBlank` here. Just use `schema.pattern` as conditions. It's more suited for this case

```javascript
const format = schema.pattern ? 'pattern' : (schema.format || 'default');
```